src/errors.rs

```diff
@@ -60,6 +61,7 @@ impl fmt::Display for Error {
             Seek => write!(formatter, "failed to seek in file `{}`", path),
             Write => write!(formatter, "failed to write to file `{}`", path),
             Flush => write!(formatter, "failed to flush file `{}`", path),
+            ReadDir => write!(formatter, "failed to read dir `{}`", path),
```

Comment:
I'm not sure if there's a convention around abbreviations in error messages, but this might be a tiny bit more readable
```suggestion
            ReadDir => write!(formatter, "failed to read directory `{}`", path),
```